controller/src/lib.rs

```diff
@@ -0,0 +1,22 @@
+#[cfg(test)]
+mod tests {
+    use std::{fs, path::PathBuf};
+
+    use ndarray::prelude::*;
+    use ndarray_npy::{ReadNpyExt, WriteNpyExt};
+
+    #[test]
+    fn read_numpy_array() {
+        let fixture = test_data_dir().join("arange(5.0).npy");
+        let file = fs::File::open(fixture.as_path()).unwrap();
+
+        let data = Array1::<f64>::read_npy(file).unwrap();
+        assert_eq!(data, Array1::range(0.0, 5.0, 1.0));
```

Comment:
testing the writing part turned out to be cumbersome: ndarray_npy adds an extra comma to the header, which is valid from the spec's point of view, but makes `write . read != id`